### ISSUE TITLE
Output config set to "When State Changes" but still sending messages minutely

### ISSUE THREAD

**Issue by jmesberg** [niklaswall/node-red-contrib-light-scheduler#14]
Pretty self explanatory: 

1/24/2018, 6:16:39 PMnode: bf5af8c9.e6d0a
msg.payload : Object
{ service: "turn_off" }
1/24/2018, 6:17:39 PMnode: bf5af8c9.e6d0a
msg.payload : Object
{ service: "turn_off" }

**Comment by jvandewiel** [comment]
There is a check on line 69 to determine if something will be sent or not based on the current `msg.payload` and a `msg.prevPayload`. This value is re-assigned *after* the msg is sent, which changes the message payload (for some reason) and makes the evaluation fail consistently and results in sending out every minute. Note that this seems only to happen if the output message is not a boolean but e.g. a string. Changing the code as per below solves the problem for me.

**Original code**
``` javascript
// Only send anything if the state have changed.
if(node.outputfreq == 'output.minutely' || msg.payload !== node.prevPayload) {
    node.send(msg);
    node.prevPayload = msg.payload;
}
```
**Modified code**
``` javascript
// Only send anything if the state have changed.
if(node.outputfreq == 'output.minutely' || msg.payload !== node.prevPayload) {
    node.prevPayload = msg.payload;
    node.send(msg);
}
```

**Comment by jmesberg** [comment]
I’m not really a coder. Can you explain how you were able to use this modified node in Node-RED before niklaswall updates his source? I can of course copy the code and make your update but how do you then create the node?

**Comment by jvandewiel** [comment]
Hi, neither am I ;-) but it is pretty easy.

In your local installation of node-red, open the file
../node_modules/node-red-contrib-light-scheduler/light-scheduler.js in a
text editor and swap the lines 71 and 72 (assuming that this is the same as
the source on github!)
Save the file and restart node-red, and it should work.

**Comment by jmesberg** [comment]
Will give it a shot! Thank you. 

John

**Comment by WizBangCrash** [comment]
@nicklaswall Great node :-) Really useful and easy to setup.
I'm seeing this same issue.
Thanks @jvandewiel for finding the offending line in the code. Just made the change and all is working as expected.

**Comment by WizBangCrash** [comment]
Okay, so I did a quick debug session my lunch hour and this is what I think the fix should be:

```
      if(node.outputfreq == 'output.minutely' || node.prevPayload === null || msg.payload.on !== node.prevPayload.on)
      {
        node.send(msg);
        node.prevPayload = msg.payload;
      }
    }

```

It is the `_msgid` that is causing the comparison to fail, so I changed to check for `null` and then check the payload value itself.

**Comment by Hugne** [comment]
Tried both of the suggested fixes but no..
This bug makes light scheduler useless if you want to override the schedule with a hw light switch (like the tradfri remote control)

**Comment by niklaswall** [comment]
This problem is solved with PR #26.
